### ISSUE TITLE
Expression resolution for default values of optional params

### ISSUE THREAD

**Issue by wqweto** [WaynePhillipsEA/twinbasic#230]
**Describe the bug**
Is seems default values expressions are resolved on invocation within the scope of the caller and are not precalculated at declaration time.

**To Reproduce**
```
Class cOptional
    Private Const STR_DEFAULT             As String = "Test"
    
    Public Sub Test(Optional MyParam As String = STR_DEFAULT)
        
    End Sub
   
End Class

Module TestModule
    Private Const STR_DEFAULT             As String = "Test"   '<---- comment this for COMPILATION ERROR

    Sub Main()
        Dim obj As New cOptional
        
        obj.Test              
        MsgBox "Done"
    End Sub
End Module
```
But the constant `STR_DEFAULT` is private for `cOptional` and missing within the scope of `Sub Main` so it cannot be resolved.

In the actual project this was encountered I got `undefined STR_DEFAULT` explanation at `obj.Test` callsite from the compiler but for the minimal  repro it just crashed with `COMPILATION ERROR` there.

TB Version: 0.10.1844

**Comment by wqweto** [comment]
Just updated the repro above to actually include a separate `STR_DEFAULT` private const in `TestModule` which allowed for `obj.Test` call to compile with optional's default value used.

**Comment by WaynePhillipsEA** [comment]
This is now fixed, in v0.10.5646.  Finally got to it :)  Thanks
